src/main/java/pages/cartPage/ShoppingCartTable.java

```diff
@@ -0,0 +1,282 @@
+    public By cartUpdateButton = By.id("cart_update");
+    public By cartCheckoutButton = By.id("cart_checkout1");
+
+    public ShoppingCartTable() {
+        super("//div[contains(@class, 'product-list')]/table");
+    }
+
+    public List<ProductModel> getProducts() {
+        List<ProductModel> products = new ArrayList<>();
+        int rowCount = getRowCount();
+
+        for (int i = 2; i <= rowCount; i++) {
+            products.add(createProductFromRow(i));
+        }
+
+        return products;
+    }
+
+    private ProductModel createProductFromRow(int rowNumber)
+    {
+        String imageUrl = getCellContent(rowNumber, getColumnNumber("Image"));
+        String name = getCellContent(rowNumber, getColumnNumber("Name"));
+        String model = getCellContent(rowNumber, getColumnNumber("Model"));
+        Currency currency = getCurrency();
+        BigDecimal unitPrice = getUnitPrice(rowNumber);
+
+        return new ProductModel(name, model, currency, unitPrice, null, null, null, imageUrl);
+    }
+
+    public int getRowIndex(String text) {
+        WebElement table = locateElement(tableLocator);
+        List<WebElement> rows = table.findElements(By.tagName("tr"));
+
+        for (int i = 0; i < rows.size(); i++) {
+            WebElement row = rows.get(i);
+            if (row.getText().contains(text)) {
+                return i;
+            }
+        }
+        return -1;
+    }
+
+    public int getRowNumber(String text) {
+        int rowIndex = getRowIndex(text);
+        if (rowIndex == -1) {
+            return -1;
+        }
+        return rowIndex + 1;
+    }
+
+    private WebElement getRow(int rowNumber) {
+        if (rowNumber == 1) {
+            throw new IllegalArgumentException("Row number 1 is a header row, not valid for this operation");
+        }
+        WebElement table = locateElement(tableLocator);
+        List<WebElement> rows = table.findElements(By.tagName("tr"));
+
+        if (rowNumber > 0 && rowNumber < rows.size()) {
+            return rows.get(rowNumber - 1);
+        } else {
+            throw new IllegalArgumentException("Row number [" + rowNumber + "] is out of range");
+        }
+    }
+
+    private WebElement getCell(int rowNumber, int columnNumber) {
+        if (rowNumber == 1) {
+            throw new IllegalArgumentException("Row number 1 is a header row, not valid for this operation");
+        }
+        List<WebElement> productRows = driver.findElements(By.xpath(tableXpath + "//tr[.//td]"));
+
+        if (rowNumber > 0 && rowNumber <= productRows.size() + 1) {
+            WebElement targetRow = productRows.get(rowNumber - 2);
+            List<WebElement> cells = targetRow.findElements(By.tagName("td"));
+
+            if (columnNumber > 0 && columnNumber <= cells.size()) {
+                return cells.get(columnNumber - 1);
+            } else {
+                throw new IllegalArgumentException("Column number [" + columnNumber + "] is out of range");
```

Comment:
Are you sure you need this exception? I think you will get it automatically if you will get out of range :)

Reply:
U are right. Unnecessary exceptions have been removed